web/webkit/src/main/scala/net/liftweb/http/LiftRules.scala

```diff
@@ -915,6 +915,118 @@ class LiftRules() extends Factory with FormVendor with LazyLoggable {
     ajaxEnd = Full(f: () => JsCmd)
   }
 
+  private [http] val ajaxAddMetaDefaultFunc:JsVar => JsCmd = toSend => Noop
+  private [http] val ajaxQueueSortDefaultFunc:JsVar => JsCmd = ajaxQueue => JE.Call("lift.ajaxQueueSortDefault", ajaxQueue)
+
+  /**
+    * Sets the `ajaxAddMeta()` and `ajaxQueueSort()` javascript functions, allowing the application's ajax
+    * retry logic to be configured.
+    *
+    * By default, Lift will retry failed ajax requests with an exponential backoff (see `LiftRules.ajaxCalcRetryTimeFunc`),
+    * allowing messages to arrive at the server in a shuffled order during periods of failed communications with the server.
+    *
+    * Alternatively, you can have Lift never send messages out of order:
+    * {{{
+    * LiftRules.ajaxAddMetaAndQueueSortFuncs.default.set(() => LiftRules.ajaxAddMetaAndQueueSortInOrderReceivedFuncs)
+    * }}}
+    *
+    * You could also define your own message ordering policy.
+    *
+    * Note: The function you provide is run in the context of a request, allowing you to have conditional logic for
+    * setting these functions.  For instance, you could have on URL which serves an application requiring ordered
+    * messages, while others do not.  Example:
+    * {{{
+    * LiftRules.ajaxAddMetaAndQueueSortFuncs.default.set { () =>
+    *   S.request match {
+    *     case Full(Req("ordered" :: Nil, _, _)) => LiftRules.ajaxAddMetaAndQueueSortInOrderReceivedFuncs
+    *     case _ => LiftRules.ajaxAddMetaAndQueueSortDefaultFuncs
+    *   }
+    * }
+    * }}}
+    *
+    */
+  val ajaxAddMetaAndQueueSortFuncs: FactoryMaker[Box[(JsVar => JsCmd, JsVar => JsCmd)]] = new FactoryMaker[Box[(JsVar => JsCmd, JsVar => JsCmd)]]( () =>
+    Full((ajaxAddMetaDefaultFunc, ajaxQueueSortDefaultFunc))
+  ) {}
+
```

Comment:
Right. The ajax queue contains all of the requests wrapped with meta data.  Hence `ajaxQueueSort()` works on those wrappers.  If you want to do something different in `ajaxQueueSort()`, then it's highly likely you'll need to add to that wrapper in `ajaxAddMeta()`.

Furthermore, the out-of-the-box configuration alternative we're offering here is the retry in order, which certainly needs both in order to function correctly.  Hence I made the pair a single setting so it's not easy to set it incorrectly.

Another alternative I considered was having these two as separate settings and provide a `LiftRules` function of type `Unit => Unit` which sets both correctly for retrying in order.  I'm not aware of a prior precedent for this approach in `LiftRules`, though.